app/controllers/v0/id_card_attributes_controller.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+require 'vic/url_helper'
+require 'vic/id_card_attribute_error'
+
+module V0
+  class IdCardAttributesController < ApplicationController
+    before_action :authorize
+
+    def show
+      id_attributes = IdCardAttributes.for_user(current_user)
+      vic_url = VIC::URLHelper.generate_url(id_attributes)
+      render json: { 'redirect' => vic_url }
+    rescue => e
+      # Catch all potential errors looking up military service history
+      # Monitor sentry to make sure this is not catching more general errors
+      log_exception_to_sentry(e)
+      raise VIC::IDCardAttributeError, status: 502, code: 'VIC011',
+                                       detail: 'Could not verify military service attributes'
+    end
+
+    private
+
+    def authorize
+      raise Common::Exceptions::Forbidden, detail: 'You do not have access to ID card attributes' unless
+        current_user.loa3?
+      raise VIC::IDCardAttributeError, status: 403, code: 'VIC002', detail: 'Unable to verify EDIPI' unless
+        current_user.edipi.present?
+      begin
+        unless current_user.veteran?
+          raise VIC::IDCardAttributeError, status: 403, code: 'VIC003',
+                                           detail: 'Not eligible for a Veteran ID Card'
+        end
+      rescue => e
+        # current_user.veteran? above may raise an error if user was not found or backend service was unavailable
+        log_exception_to_sentry(e)
+        raise VIC::IDCardAttributeError, status: 403, code: 'VIC010',
+                                         detail: 'Could not verify Veteran status'
+      end
```

Comment:
Do we really need this level of granularity with these errors? I would think a blanket Forbidden, with appropriate Sentry logging should suffice and then you don't need to modify Common::Exceptions::Forbidden or introduce a unique error for EDIPI not being present. My worry here is that we're going to be introducing a paradigm for EDIPI not being present that is too tightly coupled to VIC, I could see that particular error and ones like it being used elsewhere, so why not use something more generic?

Reply:
The changes to the Forbidden class can be reverted - I'm using it for the loa3 case for consistency with other controllers, but don't need a custom code there. And the vic-specific code and error class are subject to refactoring but for now they are consisting with the MHVAC-specific error we introduced a while back (and which I know you refactored, but we never landed that PR). 

As for granularity, I think it's warranted - I think API should be getting more granular where it can, and then we have the option of how to present to user on the front end. A blanket Forbidden doesn't give us that flexibility. This application seems especially sensitive because I hate the idea of us telling a veteran that we don't think they're a veteran - better to be specific that we can't look up their EDIPI or we are having trouble verifying their status due to a technical issue. 

Also logging to Sentry means we have to be in the path of any support issues if people call the help desk. Versus if we give them a specific error message about EDIPI being missing, then the help desk can potentially resolve it by figuring out why that user doesn't have an EDIPI. Why insert ourselves into the process and force us to look up stuff in Sentry for the help desk?